tensorflow/python/tools/optimize_for_inference_lib.py

```diff
@@ -531,3 +534,246 @@ def fuse_resize_and_conv(input_graph_def, output_node_names):
 
   result_graph_def.node.extend(new_ops)
   return result_graph_def
+
+def fuse_decomposed_batch_norm(input_graph_def):
+  """Fuse individual ops in batch normalization to FusedBatchNorm.
+
+  In some models, the batch normalizatin is performed via a group of individual
+  ops instead of using single FusedBatchNorm op. This function identifies a
+  pattern of batch normalization subgraph which is made of multiple ops and
+  transforms the graph by replacing those individual ops with FusedBatchNorm op.
+  This will provide the opportunity to further fold the FusedBatchNorm with
+  convolution ops to reduce the computation steps during inference.
+  This function currently only recognizes batch normalization patterns described
+  below, this could be extended if newer patterns are seen. Also, only target
+  for NHWC formatted graph.
+
+  Computation function:
+    (X * multiplier) + (Beta - Mean * multiplier)
+      where multiplier = rsqrt (Variance + Epsilon) * Gamma
+                    OR = rsqrt (Variance + Epsilon) when Gamma is 1
+
+  Subgraph:
+  {"Add"
+      {{"Mul"          // mul_0
+          {{"*"},     // input to apply batch normalization
+           {"Mul"      // mul_1, same op is used inside the Sub block
+              {{"Rsqrt"
+                  {"Add"
+                      {{"Const"},  //Variance
+                       {"Const"}   //Epsilon
+                      }
+                  }
+                }, // end - Rsqrt
+                {"Const"}  //Gamma
+              }
+            } // end - mul_1
+          }
+       }, // end - mul_0
```

Comment:
Nit: Please have two spaces before `//` and one space after `//`, e.g., `}**//*Text` where `s` is space. Same for other places.
```suggestion
       },  // end - mul_0
```

Reply:
will do

Reply:
Updated in the new commit.